agent/eventhandler/task_handler.go

```diff
@@ -387,6 +410,32 @@ func (taskEvents *taskSendableEvents) sendChange(change *sendableEvent,
 		// If a send event is not already in progress, trigger the
 		// submitTaskEvents to start sending changes to ECS
 		taskEvents.sending = true
+
+		//If network connection is successfully up, keep track of how many
+		// task events are being sent in order to avoid hitting the throttlingLimit.
+		if cfg.DisconnectCapable.Enabled() && !cfg.GetDisconnectModeEnabled() {
+			if handler.taskCount == 0 {
+				logger.Debug("Starting taskCountTimer.")
+				handler.taskCountTimer = time.NewTimer(time.Duration(throttlingTimeOut))
+			}
+			handler.taskCount++
+			if handler.taskCountTimer != nil && handler.checkTaskCountTimer() {
+				logger.Debug("taskCountTimer completed.")
```

Comment:
nit. could we add more contexts to this log message? such as the wait time, task event counts, and what is the next step (resetting timer and counter).